api/src/infrastructure/kubernetes/infrastructure.rs

```diff
@@ -522,8 +522,22 @@ impl Infrastructure for KubernetesInfrastructure {
         )
         .await?;
 
+        let deployment_unit_service_names = deployment_unit
+            .services()
+            .iter()
+            .map(|s| s.service_name())
+            .collect::<HashSet<_>>();
+
         let services = self.get_services_of_app(app_name).await?;
-        k8s_deployment_unit.filter_by_instances_and_replicas(&services);
+
+        k8s_deployment_unit.filter_by_instances_and_replicas(
+            services
+                .iter()
+                // we must not exclude the services that are provided by the deployment_unit
+                // because without that filter a second update of the service would create an
+                // additional deployment instead of updating/merging the existing one.
+                .filter(|s| !deployment_unit_service_names.contains(s.service_name())),
```

Comment:
I somehow do not follow this. This filter here effectively achieves exclusion at the end right?
Did you perhaps mean `we must exclude`?

Reply:
Yes, this looks right.